com.unity.multiplayer.mlapi/Runtime/Metrics/NetworkMetrics.cs

```diff
@@ -38,24 +46,37 @@ public class NetworkMetrics : INetworkMetrics
         private EventMetric<NamedMessageEvent> m_NamedMessageSentEvent = new EventMetric<NamedMessageEvent>("Named Message Sent");
         private EventMetric<NamedMessageEvent> m_NamedMessageReceivedEvent = new EventMetric<NamedMessageEvent>("Named Message Received");
 
+        private Dictionary<ulong, NetworkObjectIdentifier> m_NetworkGameObjects = new Dictionary<ulong, NetworkObjectIdentifier>();
+
+        private ConnectionInfo m_LocalConnectionInfo;
+
         public NetworkMetrics(NetworkManager networkManager)
         {
             m_NetworkManager = networkManager;
             m_Dispatcher = new MetricDispatcherBuilder()
                 .WithMetricEvents(m_NamedMessageSentEvent, m_NamedMessageReceivedEvent)
                 .Build();
-            
+
             m_Dispatcher.RegisterObserver(MLAPIObserver.Observer);
+            m_LocalConnectionInfo = new ConnectionInfo(m_NetworkManager.LocalClientId);
+        }
+
+        public void TrackNetworkObject(NetworkObject networkObject)
+        {
+            if (!m_NetworkGameObjects.ContainsKey(networkObject.NetworkObjectId))
+            {
+                m_NetworkGameObjects[networkObject.NetworkObjectId] = new NetworkObjectIdentifier(networkObject.name, networkObject.NetworkObjectId);
```

Comment:
I'm not sure if `NetworkObjectIdentifier` needs to be cached like this. It is a struct so creating the instance is free. Also this means that changes to the network object's name will not appear in the profiler.

Although `.name` might allocate memory since it pulls from native - in which case cashing _might_ still be preferred to avoid the allocation?

Reply:
This sort of thing add complexity, and we can only guess at the benefits. I'm a fan of keeping this stuff as simple as possible, and then optimizing when we have cold hard data to back it up.

If we have to optimize part of this, the solution might turn out to be something else entirely.

Make it work, make it clean, make it fast. In that order.

Reply:
Yes I was under that impression that caching would be beneficial if we're going to be reusing the structures a lot.

Reply:
I think the cost isn't worth the incidental complexity it adds to the code.